tools/release.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+// release.go creates a new version tag and pushes it to GitHub. It has two
+// main modes of operation:
+// * default: fetch origin/main, tag that commit with Major.Minor.Patch version
+//   of 0.YYYYMMDD.0, and push the tag.
+// * hotfix: fetch the named release tag, create a new branch starting from
+//   that commit, cherry-pick the named commit(s) onto that branch, tag the
+//   new branch head with a Major.Minor.Patch that matches the Major.Minor of
+//   named release tag and a Patch that is one greater, and push that tag.
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"time"
+
+	"golang.org/x/mod/semver"
+)
+
+func runOrDie(cmd *exec.Cmd) string {
+	fmt.Println(cmd.String())
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		fmt.Println(string(out))
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	return string(out)
+}
+
+func main() {
+	if len(os.Args) >= 2 && os.Args[1] == "hotfix" {
```

Comment:
Could you add another conditional that shows the operator a basic help menu outside of flag.Parse()? Without this, they'd have to read the code to see that hotfix is an available subcommand.
```suggestion
	if os.Args[1] == "-h" {
		fmt.Println("go run release.go [hotfix] [args]\n" +
			"-h : display help text")
	} else if len(os.Args) >= 2 && os.Args[1] == "hotfix" {
```